kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-kogito/kie-wb-common-stunner-kogito-client/src/main/java/org/kie/workbench/common/stunner/kogito/client/session/command/impl/KogitoKeyEventHandlerImpl.java

```diff
@@ -52,63 +53,79 @@
+            registerShortcut(keyboardShortcutsApi.registerKeyPress(
                     asStringCodes(kogitoShortcutCallback.getKeyCombination()),
                     kogitoShortcutCallback.getLabel(),
                     target -> runIfEnabled(shortcutCallback::onKeyShortcut),
-                    kogitoShortcutCallback.getOpts()));
+                    kogitoShortcutCallback.getOpts()), shortcutCallback);
         }
 
         //Session
         else if (shortcutCallback instanceof SessionKeyShortcutCallback && ((SessionKeyShortcutCallback) shortcutCallback).getDelegate() instanceof KogitoKeyShortcutKeyDownThenUp) {
-            registeredShortcutsIds.add(keyboardShortcutsApi.registerKeyDownThenUp(
+            registerShortcut(keyboardShortcutsApi.registerKeyDownThenUp(
                     asStringCodes(kogitoShortcutCallback.getKeyCombination()),
                     kogitoShortcutCallback.getLabel(),
                     target -> runIfEnabled(shortcutCallback::onKeyShortcut),
                     target -> runIfEnabled(() -> shortcutCallback.onKeyUp(null)),
-                    kogitoShortcutCallback.getOpts()));
+                    kogitoShortcutCallback.getOpts()), shortcutCallback);
         } else if (shortcutCallback instanceof SessionKeyShortcutCallback && ((SessionKeyShortcutCallback) shortcutCallback).getDelegate() instanceof KogitoKeyPress) {
-            registeredShortcutsIds.add(keyboardShortcutsApi.registerKeyPress(
+            registerShortcut(keyboardShortcutsApi.registerKeyPress(
                     asStringCodes(kogitoShortcutCallback.getKeyCombination()),
                     kogitoShortcutCallback.getLabel(),
                     target -> runIfEnabled(shortcutCallback::onKeyShortcut),
-                    kogitoShortcutCallback.getOpts()));
+                    kogitoShortcutCallback.getOpts()), shortcutCallback);
         }
 
         //Default
         else {
-            registeredShortcutsIds.add(keyboardShortcutsApi.registerKeyPress(
+            registerShortcut(keyboardShortcutsApi.registerKeyPress(
                     asStringCodes(kogitoShortcutCallback.getKeyCombination()),
                     kogitoShortcutCallback.getLabel(),
                     target -> runIfEnabled(shortcutCallback::onKeyShortcut),
-                    KeyboardShortcutsApiOpts.DEFAULT));
+                    KeyboardShortcutsApiOpts.DEFAULT), shortcutCallback);
         }
 
         return this;
     }
 
+    private void registerShortcut(Integer registeredKey, final KeyboardControl.KeyShortcutCallback callback) {
+        registeredShortcutsIds.add(registeredKey);
+        callbackMap.put(callback, registeredKey);
+    }
+
+    @Override
+    public void removeKeyShortcutCallback(final KeyboardControl.KeyShortcutCallback shortcutCallback) {
+        final Integer key = callbackMap.get(shortcutCallback);
+
+        for (Integer registeredKey: registeredShortcutsIds) {
+            if (registeredKey == key) {
```

Comment:
imho the sonar suggestion is good here, could you please fix?

Reply:
Hi @domhanak good point, the issue is, what if for some reason, the value is null? then it would throw NPE, let me know

Reply:
@inodeman I would add a test case for that scenario, if it is actually possible.

Reply:
Hi @domhanak I went ahead and used equals with null checking so, we are ok now.